src/components/singleLineDiagram/single-line-diagram.js

```diff
@@ -953,6 +962,12 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
                     {props.updateSwitchMsg && (
                         <Alert severity="error">{props.updateSwitchMsg}</Alert>
                     )}
+                    {!allowModificationsOnNode(workingNode, selectedNode) &&
+                        selectedNode?.type !== 'ROOT' && (
```

Comment:
Same as before, maybe use a function whose name makes sense with the warning displayed.